### ISSUE TITLE
override block in post using nunjucks

### ISSUE THREAD

**Issue by btrem** [11ty/eleventy#1632]
If I have a block defined in a layout using nunjucks, can I override it in a post? If this is my layout:

    
    {% block content %}
        <main>
            {{ content | safe }}
        </main>
    {% endblock %}

In mypost.njk, can I do this?

    ---
    title: my post
    ---
    {% block content %}
        <main class="specialClassForMyPost">
            <p>This is my post.</p>
        </main>
    {% endblock %}

So in other words change the `main` element on a one off basis. That gives me duplicate `<main>` tags, which means either it can't be done or I'm doing it wrong.

**Comment by Ryuno-Ki** [comment]
Sounds like a use case for [`{% extends %}`](https://mozilla.github.io/nunjucks/templating.html#extends) to me :thinking:

**Comment by pdehaan** [comment]
@btrem Do you have a public repo somewhere? I might be able to take a look.

https://github.com/11ty/eleventy/issues/853 might also have some good pointers and links to examples in repos.

**Comment by btrem** [comment]
Sorry, of course I messed up my initial question. I edited it to make it clear what I'm hoping to achieve.

@Ryuno-Ki So it sounds like I *can't* override a block in a a post as I demonstrated in the edited initial post. Instead, I'd need to create a new layout, say, `my-new-layout.njk`:

    {% extends 'base.njk' %}
    {% block content %}
        <main class="specialClassForMyPost">
            {{ content | safe }}
        </main>
    {% endblock %}

and then in my-post.njk (or my-post.md or whatever):

    ---
    title: my post
    layout: my-new-layout.njk
    ---

    <main class="specialClassForMyPost">
        <p>This is my post.</p>
    </main>

**Comment by btrem** [comment]
@pdehaan Thanks for the link to the github thread. It confirms what I thought -- and provides some interesting reading -- which is not a big deal. I can create a layout that extends my base layout.
